aws-s3-transfer-manager/src/operation/download/object_meta.rs

```diff
@@ -45,13 +38,18 @@ pub struct ObjectMetadata {
     pub ssekms_key_id: Option<String>,
     pub bucket_key_enabled: Option<bool>,
     pub storage_class: Option<aws_sdk_s3::types::StorageClass>,
-    pub request_charged: Option<aws_sdk_s3::types::RequestCharged>,
     pub replication_status: Option<aws_sdk_s3::types::ReplicationStatus>,
     pub parts_count: Option<i32>,
-    pub tag_count: Option<i32>,
     pub object_lock_mode: Option<aws_sdk_s3::types::ObjectLockMode>,
     pub object_lock_retain_until_date: Option<::aws_smithy_types::DateTime>,
     pub object_lock_legal_hold_status: Option<aws_sdk_s3::types::ObjectLockLegalHoldStatus>,
+
+    /// The request_id if the client made a request to retrieve object metadata, such as with HeadObject.
+    pub request_id: Option<String>,
```

Comment:
Also related is this [todo](https://github.com/awslabs/aws-s3-transfer-manager-rs/blob/c88e07f7c92a3fe8bd391c624876538743c5314b/aws-s3-transfer-manager/src/error.rs#L148) and that we need to expose metadata for errors as well

Reply:
Thanks, I have updated it to use the same trait. I have not looked into exposing metadata for errors yet. I will keep it as a TODO for now for a future PR, since I want to get this merged first to unblock other people.